cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/batch/sources/DBSource.java

```diff
@@ -0,0 +1,81 @@
+import co.cask.cdap.templates.etl.api.StageConfigurer;
+import co.cask.cdap.templates.etl.api.batch.BatchSource;
+import co.cask.cdap.templates.etl.api.batch.BatchSourceContext;
+import co.cask.cdap.templates.etl.common.DBRecord;
+import co.cask.cdap.templates.etl.common.Properties;
+import com.google.common.base.Preconditions;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.LongWritable;
+import org.apache.hadoop.mapreduce.Job;
+import org.apache.hadoop.mapreduce.lib.db.DBConfiguration;
+import org.apache.hadoop.mapreduce.lib.db.DBInputFormat;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+/**
+ * Batch source to read from a Database table
+ */
+public class DBSource extends BatchSource<LongWritable, DBRecord> {
+  private static final Logger LOG = LoggerFactory.getLogger(DBSource.class);
+
+  @Override
+  public void configure(StageConfigurer configurer) {
+    configurer.setName("DBBatchSource");
+    configurer.setDescription("Batch source for database");
+    configurer.addProperty(new Property(Properties.DB.DB_DRIVER_CLASS, "Driver class to connect to the database", true));
+    configurer.addProperty(new Property(Properties.DB.DB_CONNECTION_STRING, "JDBC connection string including database name", true));
+    configurer.addProperty(new Property(Properties.DB.DB_TABLE_NAME, "Table name to import", true));
+    configurer.addProperty(new Property(Properties.DB.DB_COLUMNS, "Comma-separated list of columns to import", false));
+  }
+
+  @Override
+  public void prepareJob(BatchSourceContext context) {
+    Map<String, String> runtimeArguments = context.getRuntimeArguments();
+    String dbDriverClass = runtimeArguments.get(Properties.DB.DB_DRIVER_CLASS);
+    String dbConnectionString = runtimeArguments.get(Properties.DB.DB_CONNECTION_STRING);
+    String dbTableName = runtimeArguments.get(Properties.DB.DB_TABLE_NAME);
+    String dbColumns = runtimeArguments.get(Properties.DB.DB_COLUMNS);
+    Preconditions.checkArgument(dbDriverClass != null, "dbDriverClass cannot be null");
+    Preconditions.checkArgument(dbConnectionString != null, "dbConnectionString cannot be null");
+    Preconditions.checkArgument(dbTableName != null, "dbTableName cannot be null");
+
+    LOG.debug("dbTableName = {}; dbDriverClass = {}; dbConnectionString = {}; dbColumns = {}",
+              dbTableName, dbDriverClass, dbConnectionString, dbColumns);
+
+    String countQueryColumn;
+    if (dbColumns == null) {
+      dbColumns = "*";
+      countQueryColumn = "*";
+    } else {
+      countQueryColumn = dbColumns.substring(0, dbColumns.indexOf(","));
+    }
+
+    Job job = context.getHadoopJob();
+    Configuration conf = job.getConfiguration();
+    job.setInputFormatClass(DBInputFormat.class);
+    DBConfiguration.configureDB(conf, dbDriverClass, dbConnectionString);
+    DBInputFormat.setInput(job, DBRecord.class, String.format("SELECT %s FROM %s", dbColumns, dbTableName),
```

Comment:
Regarding sql injection, there is another API that ``DBInputFormat`` exposes. However, the signature for that is - 

```
public static void setInput(Job job,
                            Class<? extends DBWritable> inputClass,
                            String tableName,
                            String conditions,
                            String orderBy,
                            String... fieldNames)
```
We could use that, but it would enforce the user to always specify all columns (to send to ``fieldNames``. Thoughts?

However, even this format accepts conditions that get substituted in the ``WHERE`` clause.